src/angular/components/sso.component.ts

```diff
@@ -53,13 +53,32 @@ export class SsoComponent {
                 if (qParams.code != null && codeVerifier != null && state != null && state === qParams.state) {
                     await this.logIn(qParams.code, codeVerifier);
                 }
+                const clientId = await this.storageService.get<string>(ConstantsService.ssoClientId);
+                if (clientId === ConstantsService.browserClientId) {
+                    // Prevent entering this if we're not in the web vault.
+                    if (location.protocol === 'https://') {
+                        window.postMessage({ type: "AUTH_RESULT", code: qParams.code, state: qParams.state }, "*");
+                        this.router.navigate(['sso-complete']);
+                    }
+                }
             } else if (qParams.clientId != null && qParams.redirectUri != null && qParams.state != null &&
                 qParams.codeChallenge != null) {
                 this.redirectUri = qParams.redirectUri;
                 this.state = qParams.state;
                 this.codeChallenge = qParams.codeChallenge;
                 this.clientId = qParams.clientId;
             }
+
+            if (qParams.clientId == ConstantsService.browserClientId) {
```

Comment:
I think we could just piggyback on the qParams.redirectUri to get the desired functionality without having to introduce a new param. Let's have a call.